pkg/scaling/scale_handler.go

```diff
@@ -140,8 +140,9 @@ func (h *scaleHandler) startScaleLoop(ctx context.Context, withTriggers *kedav1a
 	pollingInterval := getPollingInterval(withTriggers)
 	logger.V(1).Info("Watching with pollingInterval", "PollingInterval", pollingInterval)
 
-	tmr := time.NewTimer(pollingInterval)
 	for {
+		tmr := time.NewTimer(pollingInterval)
+
```

Comment:
@zroubalik this is a subtle fix (yet obviously important!) - is it possible somehow to add a test for it? the only apparent way to me would be to pull the timer-related code out into a utility function and then test that. thoughts?

Reply:
(and if so, I can pick that up in a follow-up PR)

Reply:
I am concerned that pulling that part of code to the utility function would make the code less readable and complex. WDYT? 
Anyway more tests are always good, so if you come up with some good way how to test this stuff, I wouldn't be mad :)

Reply:
ha, that was my idea to make it more testable. I don't have others at the moment, but I'll keep my mind open 👍